### PR TITLE
Add fuzzing targets for standard library and qrcode

### PR DESCRIPTION
**Pull request by bboilot-ledger**
## Description

This MR adds fuzzing targets for the following functions:
- apdu_parser
- base58_decode
- bip32_path_read
- qrcodegen_encodeBinary

It also fixes a read out of bounds in bip32_path_read that was found by the fuzzer.

This is related to https://github.com/LedgerHQ/ledger-secure-sdk/issues/25

## Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [X] Tests
- [ ] Documentation
- [ ] Other (for changes that might not fit in any category)

### CODE REVIEW

File: fuzzing/fuzzer_apdu_parser.c [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+
+#include "parser.h"
+#include "base58.h"
+#include "bip32.h"
+#include "qrcodegen.h"
```
Comment (xchapron-ledger) [inline fuzzing/fuzzer_apdu_parser.c:8]:
```suggestion
```
Unused?

File: fuzzing/fuzzer_apdu_parser.c [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#include "os_io.h"
```
Comment (xchapron-ledger) [inline fuzzing/fuzzer_apdu_parser.c:9]:
```suggestion
#define IO_APDU_BUFFER_SIZE (5 + 255)
```
and delete the `os_io.h` file?

File: fuzzing/fuzzer_base58.c [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+
+#include "parser.h"
+#include "base58.h"
+#include "bip32.h"
+#include "qrcodegen.h"
```
Comment (xchapron-ledger) [inline fuzzing/fuzzer_base58.c:8]:
```suggestion
#include "base58.h"
```

File: fuzzing/fuzzer_bip32.c [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+
+#include "parser.h"
+#include "base58.h"
+#include "bip32.h"
+#include "qrcodegen.h"
```
Comment (xchapron-ledger) [inline fuzzing/fuzzer_bip32.c:8]:
```suggestion
#include "bip32.h"
```

File: fuzzing/fuzzer_qrcodegen.c [hunk 5]
```diff
@@ -0,0 +1,27 @@
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+
+#include "parser.h"
+#include "base58.h"
+#include "bip32.h"
```
Comment (xchapron-ledger) [inline fuzzing/fuzzer_qrcodegen.c:7]:
```suggestion
```

File: fuzzing/fuzzer_apdu_parser.c (hunk 1 above)
Comment (bboilot-ledger) [reply fuzzing/fuzzer_apdu_parser.c]: Yes, I removed unused headers

File: fuzzing/fuzzer_apdu_parser.c (hunk 2 above)
Comment (bboilot-ledger) [reply fuzzing/fuzzer_apdu_parser.c]: Yes, I removed unused headers and directly set `IO_APDU_BUFFER_SIZE` in `fuzzer_apdu_parser.c`.

**Review by xchapron-ledger — APPROVED** [review]

File: fuzzing/fuzzer_base58.c (hunk 3 above)
Comment (bboilot-ledger) [reply fuzzing/fuzzer_base58.c]: Done. And I also removed unused standard headers.

File: fuzzing/fuzzer_bip32.c (hunk 4 above)
Comment (bboilot-ledger) [reply fuzzing/fuzzer_bip32.c]: Same here.

File: fuzzing/fuzzer_qrcodegen.c (hunk 5 above)
Comment (bboilot-ledger) [reply fuzzing/fuzzer_qrcodegen.c]: And here.

**Pull request merged**